api/src/routing.js

```diff
@@ -12,15 +12,21 @@ const prometheusMiddleware = require('prometheus-api-metrics');
 const rateLimiterMiddleware = require('./middleware/rate-limiter');
 const logger = require('./logger');
 const isClientHuman = require('./is-client-human');
-const target = 'http://' + environment.host + ':' + environment.port;
-const proxy = require('http-proxy').createProxyServer({ target: target });
+
+const port = typeof environment.port !== 'undefined' && environment.port !== null ? `:${environment.port}` : '';
+const target = `${environment.protocol}//${environment.host}${port}`;
+const proxy = require('http-proxy').createProxyServer({
+  target: target, changeOrigin: environment?.proxies?.default?.changeOrigin ?? false 
```

Comment:
Nit: For formatting let's put each value on its own line just like the two proxies below...

```suggestion
  target,
  changeOrigin: environment?.proxies?.default?.changeOrigin ?? false 
```

Reply:
@garethbowen have you ever considered adding prettier to the project?

Reply:
Yes. It's very challenging to add it to an existing project because every time you modify a file it makes a bunch of unrelated changes that make the review difficult. Furthermore the developers agreed it actually made the code less readable so we removed it.

Reply:
It only makes unrelated changes if it isn't enforced in CI?

I've moved projects over to it. You have a single format change and then there's no noise on PRs. 

In fact, the main reasons I use it is to a) reduce noise on PRs b) avoid formatting issues coming up/being discussed in review. 

If IDE integrated on save, it does have another advantage, which is that you get a snap to format on code that is structurally valid, so it's really great for hunting for closing brackets quickly. 

I agree it's not always the best defaults (the ternary formatting is bad, for example), but there are few things I love less than debating formatting on review. 

Speaking of which, going to make the requested change.